src/posts/posts.service.ts

```diff
@@ -101,8 +141,20 @@ export class PostsService {
   }
   async getSinglePost(postId: string): Promise<Post> {
     const post = await this.postRepository.getSinglePost(postId);
+
+    // check whether post is found
+    if (!post) throw new NotFoundException(`Post with id: ${postId} not found`);
+
+    // don't return post if post.created = false
+    if (!post.created) {
+      throw new HttpException(
+        `Post with id: ${postId} still under creation...`,
+        423,
```

Comment:
Please use either: 
1. extend `HttpStatus` from nest
2. extend nest expectations and introduce `LockedExpection`
behind the scene, please create `PickifyHttpStatus` enum (extending `HttpStatus` from `@nestjs/commons` or at least create our own Enum

Reply:
Both changes are added:
1- extended `HttpStatus` using `PickifyHttpStatus`
2- Added `LockedException`